connectors/sources/sharepoint_online.py

```diff
@@ -1772,6 +1812,40 @@ async def site_list_items(
             ]:  # TODO: make it more flexible. For now I ignore them cause they 404 all the time
                 continue
 
+            has_unique_role_assignments = False
+
+            if self.configuration["fetch_unique_list_item_permissions"]:
+                has_unique_role_assignments = (
+                    await self.client.site_list_item_has_unique_role_assignments(
+                        site_web_url, site_list_name, list_item_natural_id
+                    )
+                )
+
+                if has_unique_role_assignments:
+                    self._logger.debug(
+                        f"Fetching unique permissions for list item with id '{list_item_natural_id}'. Ignoring parent site permissions."
+                    )
+
+                    list_item_access_control = []
+
+                    async for role_assignment in self.client.site_list_item_role_assignments(
+                        site_web_url, site_list_name, list_item_natural_id
+                    ):
+                        list_item_access_control.extend(
+                            await self._get_access_control_from_role_assignment(
+                                role_assignment
+                            )
+                        )
+
+                    list_item = self._decorate_with_access_control(
+                        list_item, list_item_access_control
+                    )
+
+            if not has_unique_role_assignments:
```

Comment:
This could just be an `else:` and you could remove the first definition of `has_unique_role_assignments = False` above.

Reply:
Yup, I shamelessly copy-pasted this from other place in the code-base. Will fix both.